### PR TITLE
[#551] docs: update templates for flaky test and pull request

### LINKED ISSUE #551: [Improvement][Style] add autolink to issues in PR title and commit msg
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-uniffle/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

It's common and considered best practice to be able to click links to corresponding issues/PRs/tickets in the commit msg. To achieve this features, Github's autolinked references and links[^1] are used. 

For example, Spark uses this feature to autolink Jiras. 
image

Since Uniffle are using Github Issues to track issues/request/improvements, etc. The issue is effective the same as the Jira ticket.

I'd like to propose an suggestion to improve the code context experience: add autolink to issues in PR title and commit msg.

[^1]: https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/autolinked-references-and-urls

### How should we improve?

1. Let's decide which form to include autolinks:
   - [#xxx], the simple form, https://github.com/apache/incubator-uniffle/commit/cf9afab471a9595d8298ebd65c4d25c080ad4bbf
     image 
   - [ISSUE #xxx] prefixed with the `ISSUE`
2. Enforce the pr title check.  Every PR should have an issue linked. 
    There's an exception rule for this: PR title prefixed with [Minor]/[MINOR] can be excused for this enforcement check.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by kaijchen**
### What changes were proposed in this pull request?

Update flaky test template and pr template.

### Why are the changes needed?

1. Flaky test template
    * Title is not consistent with other templates.
    * Default parent issue is invalid. 
2. PR template
    * Make the instructions shorter to read.
    * Let PR author remove the instructions so they will read (hopefully).
    * Update instructions, and add example for PR title. (Resolves #551)

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

No need.

### CODE REVIEW

**Review by jerqi — DISMISSED** [review]
LGTM, thanks @kaijchen

File: .github/PULL_REQUEST_TEMPLATE [hunk 1]
```diff
@@ -1,44 +1,34 @@
...
-<!--
-Please clarify what changes you are proposing. The purpose of this section is to outline the changes and how this PR fixes the issue.
-If possible, please consider writing useful notes for better and faster reviews in your PR. See the examples below.
-  1. If you refactor some codes with changing classes, showing the class hierarchy will help reviewers.
-  2. If you fix some SQL features, you can provide some references of other DBMSes.
-  3. If there is design documentation, please add the link.
-  4. If there is a discussion in the mailing list, please add the link.
--->
 
+Please outline the changes and how this PR fixes the issue.
```
[the remark this answers is not in the document]
Comment (kaijchen) [reply .github/PULL_REQUEST_TEMPLATE]: Is it enough to put the tips in parentheses?

File: .github/PULL_REQUEST_TEMPLATE [hunk 2]
```diff
@@ -1,44 +1,34 @@
...
-<!--
-Note that it means *any* user-facing change including all aspects such as the documentation fix.
-If yes, please clarify the previous behavior and the change this PR proposes - provide the console output, description and/or an example to show the behavior difference if possible.
-If possible, please also clarify if this is a user-facing change compared to the released versions or within the unreleased branches such as master.
-If no, write 'No'.
--->
 
+Please list the user-facing changes introduced by your change, including
+  1. change in user-facing APIs
+  2. addition or removal of property keys
```
[the remark this answers is not in the document]
Comment (kaijchen) [reply .github/PULL_REQUEST_TEMPLATE]: Added default answer: "No."

File: .github/PULL_REQUEST_TEMPLATE [hunk 3]
```diff
@@ -1,44 +1,34 @@
...
 ### Why are the changes needed?
-<!--
+
 Please clarify why the changes are needed. For instance,
   1. If you propose a new API, clarify the use case for a new API.
-  2. If you fix a bug, you can clarify why it is a bug.
--->
+  2. If you fix a bug, describe the bug.
 
+Fix: #
```
[the remark this answers is not in the document]
Comment (kaijchen) [reply .github/PULL_REQUEST_TEMPLATE:25]: This is to remind PR author to link the issue.

**Comment by kaijchen** [comment]
Thanks @advancedxy @jerqi for the review.
Although there might be some concerns, PPMC members and committers are able to edit the PR title and description.
I think we can give it a try.

File: .github/PULL_REQUEST_TEMPLATE (hunk 3 above)
[the remark this answers is not in the document]
Comment (kaijchen) [reply .github/PULL_REQUEST_TEMPLATE:25]: How to change it? How about this?

```suggestion
Fix: #(issue id)
```

File: .github/PULL_REQUEST_TEMPLATE [hunk 4]
```diff
@@ -1,44 +1,39 @@
 <!--
-Thanks for sending a pull request!  Here are some tips for you:
-  1. If this is your first time, please read our contributor guidelines: https://github.com/apache/incubator-uniffle/blob/master/CONTRIBUTING.md
-  2. Ensure you have added or run the appropriate tests for your PR
-  3. If the PR is unfinished, add '[WIP]' in your PR title, e.g., '[WIP]XXXX Your PR title ...'.
-  4. Be sure to keep the PR description updated to reflect all changes.
-  5. Please write your PR title to summarize what this PR proposes.
-  6. If possible, provide a concise example to reproduce the issue for a faster review.
+1. Title: [#<issue>]<type>(<scope>): <subject>
```
Comment (kaijchen) [inline .github/PULL_REQUEST_TEMPLATE:2]:
```suggestion
1. Title: [#<issue>] <type>(<scope>): <subject>
```

File: .github/PULL_REQUEST_TEMPLATE [hunk 5]
```diff
@@ -1,44 +1,39 @@
...
+   Examples:
+     - "[#123] feat(operator): support xxx"
+     - "[#233] fix: check null before access result in xxx"
+     - "[MINOR] refactor: fix typo in variable name"
+     - "[MINOR] docs: fix typo in README"
+     - "[#255] test: fix flaky test NameOfTheTest"
+   Reference: https://gist.github.com/joshbuchea/6f47e86d2510bce28f8e7f42ae84c716
```
[the remark this answers is not in the document]
Comment (kaijchen) [reply .github/PULL_REQUEST_TEMPLATE:9]: why not?

File: .github/PULL_REQUEST_TEMPLATE [hunk 6]
```diff
@@ -1,44 +1,39 @@
...
-  5. Please write your PR title to summarize what this PR proposes.
-  6. If possible, provide a concise example to reproduce the issue for a faster review.
+1. Title: [#<issue>] <type>(<scope>): <subject>
+   Examples:
+     - "[#123] feat(operator): support xxx"
+     - "[#233] fix: check null before access result in xxx"
+     - "[MINOR] refactor: fix typo in variable name"
+     - "[MINOR] docs: fix typo in README"
+     - "[#255] test: fix flaky test NameOfTheTest"
+   Reference: https://gist.github.com/joshbuchea/6f47e86d2510bce28f8e7f42ae84c716
```
Comment (kaijchen) [inline .github/PULL_REQUEST_TEMPLATE:9]:
```suggestion
   Reference: https://www.conventionalcommits.org/en/v1.0.0/
```

**Comment by kaijchen** [comment]
Thanks @advancedxy and @jerqi for the review.

File: .github/ISSUE_TEMPLATE/flaky-test-report.yml [hunk 7]
```diff
@@ -66,7 +66,7 @@ body:
   - type: textarea
     attributes:
       label: Parent issue
-      value: "#1733"
+      value: ""
```
Comment (xianjingfeng) [inline .github/ISSUE_TEMPLATE/flaky-test-report.yml:69]: `value` must be of type String and cannot be empty. @kaijchen

Comment (kaijchen) [reply .github/ISSUE_TEMPLATE/flaky-test-report.yml:69]: Thanks, can you submit a PR to fix it?
Maybe let's just remove this field?

Comment (xianjingfeng) [reply .github/ISSUE_TEMPLATE/flaky-test-report.yml:69]: ok

**Pull request merged**